### ISSUE TITLE
The parsing of the `--exclude-config` option seems to be broken, preventing `native-image` to correctly parse the rest of the command line

### ISSUE THREAD

**Issue by guizmaii** [oracle/graal#7010]
Hi

I'm trying to make this project compile my CLI app: https://github.com/gnp/zio-quickstart-https/pull/1
For this, I use [sbt-native-image](https://github.com/scalameta/sbt-native-image) and I encounter issues with the `--exclude-config` option (I don't find its documentation BTW)

To eliminate the "sbt-native-image" factor from the equation, I just copied the command line that it's generating and I copy/pasted it in my terminal.

First, using ZSH, the `--exclude-config .*.jar,.*.properties` makes ZSH tried to do something and the command fails with:
```bash
zsh: no matches found: .*.jar,.*.properties
```

To avoid this issue, I ran the same command but wrapped in `bash -c "native-image ..."`
Here's the full command:
```bash
bash -c "native-image -cp /Users/jules/jules_ivanic/workspace/zio-quickstart-https/modules/cli/target/native-image-internal/manifest.jar --exclude-config .*.jar,.*.properties --no-fallback --enable-http --enable-https --enable-url-protocols=http,https --install-exit-handlers --diagnostics-mode -Djdk.http.auth.tunneling.disabledSchemes= --initialize-at-run-time=io.netty.handler.ssl.BouncyCastleAlpnSslUtils,io.netty.channel.epoll.Epoll,io.netty.channel.epoll.Native,io.netty.channel.epoll.EpollEventLoop,io.netty.channel.epoll.EpollEventArray,io.netty.channel.DefaultFileRegion,io.netty.channel.kqueue.KQueueEventArray,io.netty.channel.kqueue.KQueueEventLoop,io.netty.channel.kqueue.KQueue,io.netty.channel.kqueue.Native,io.netty.channel.unix.Errors,io.netty.channel.unix.IovArray,io.netty.channel.unix.Limits,io.netty.util.internal.logging.Log4JLogger -H:ConfigurationFileDirectories=/Users/jules/jules_ivanic/workspace/zio-quickstart-https/modules/cli/src/main/resources/META-INF/native-image com.gregorpurdy.cli.MyCLI /Users/jules/jules_ivanic/workspace/zio-quickstart-https/my-cli"
```

And here's the resulting execution:
```bash
# Diagnostics mode enabled: image-build reports are saved to reports/diagnostics_20230718_135939
Apply jar:file:///Users/jules/.sdkman/candidates/java/20.0.1-graal/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties
Apply jar:file:///Users/jules/.sdkman/candidates/java/20.0.1-graal/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties
# Printing command line arguments to: /Users/jules/reports/diagnostics_20230718_135939/command-line_20230718_135939.txt
========================================================================================================================
GraalVM Native Image: Generating 'my-cli' (executable)...
========================================================================================================================
# Printing feature information to: /Users/jules/jules_ivanic/workspace/zio-quickstart-https/reports/diagnostics_20230718_135939/feature_info_20230718_135942.csv
Warning: Could not resolve class sun.misc.VM for reflection configuration. Reason: java.lang.ClassNotFoundException: sun.misc.VM.
Warning: Method sun.security.provider.NativePRNG.<init>(SecureRandomParameters) not found.
Warning: Could not resolve class sun.misc.VM for reflection configuration. Reason: java.lang.ClassNotFoundException: sun.misc.VM.
Warning: Method sun.security.provider.NativePRNG.<init>(SecureRandomParameters) not found.
[1/8] Initializing...                                                                                    (8.2s @ 0.33GB)
 Java version: 20.0.1+9, vendor version: Oracle GraalVM 20.0.1+9.1
 Graal compiler: optimization level: 2, target machine: x86-64-v3, PGO: off
 C compiler: cc (apple, x86_64, 12.0.5)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 1 user-specific feature(s)
 - com.oracle.svm.polyglot.scala.ScalaFeature
# Printing compilation-target information to: /Users/jules/jules_ivanic/workspace/zio-quickstart-https/reports/diagnostics_20230718_135939/target_info_20230718_135948.txt
# Printing class initialization configuration to: /Users/jules/jules_ivanic/workspace/zio-quickstart-https/reports/diagnostics_20230718_135939/class_initialization_configuration_20230718_135948.csv
# Printing native-library information to: /Users/jules/jules_ivanic/workspace/zio-quickstart-https/reports/diagnostics_20230718_135939/native_library_info_20230718_140027.txt
# Printing substitutions performed by native-image to: /Users/jules/jules_ivanic/workspace/zio-quickstart-https/reports/diagnostics_20230718_135939/substitutions_20230718_140027.csv
# Printing class initialization dependencies to: /Users/jules/jules_ivanic/workspace/zio-quickstart-https/reports/diagnostics_20230718_135939/class_initialization_dependencies_20230718_140028.dot
# Printing class initialization report to: /Users/jules/jules_ivanic/workspace/zio-quickstart-https/reports/diagnostics_20230718_135939/class_initialization_report_20230718_140028.csv
[2/8] Performing analysis...  [*****]                                                                   (39.4s @ 5.08GB)
  13,271 (87.81%) of 15,114 types reachable
  19,895 (63.18%) of 31,490 fields reachable
  81,361 (50.07%) of 162,479 methods reachable
   3,208 types,   179 fields, and 2,347 methods registered for reflection
       5 native libraries: -framework CoreServices, -framework Foundation, dl, pthread, z

Warning: Aborting stand-alone image build due to unsupported features
com.oracle.svm.hosted.FallbackFeature$FallbackImageRequest: Aborting stand-alone image build due to unsupported features
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FallbackFeature.reportFallback(FallbackFeature.java:240)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FallbackFeature.reportFallback(FallbackFeature.java:225)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FallbackFeature.afterAnalysis(FallbackFeature.java:306)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$12(NativeImageGenerator.java:788)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:86)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:788)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:582)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:539)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:408)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:612)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:134)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:94)
------------------------------------------------------------------------------------------------------------------------
                        1.6s (3.3% of total time) in 25 GCs | Peak RSS: 5.72GB | CPU load: 8.76
========================================================================================================================
Finished generating 'my-cli' in 47.9s.
Generating fallback image...
Warning: Image 'my-cli' is a fallback image that requires a JDK for execution (use --no-fallback to suppress fallback image generation and to print more detailed information why a fallback image was necessary).
```

We can see that the `--no-fallback` present in the options passed to `native-image` didn't get taken into account

**Comment by fniephaus** [comment]
To avoid shell expansion, you must quote the argument. Does `"--exclude-config=.*.jar,.*.properties"` work for you?

**Comment by guizmaii** [comment]
```bash
➜  ~ bash -c "native-image -cp /Users/jules/workspace/myproject/modules/cli/target/native-image-internal/manifest.jar \"--exclude-config=.*.jar,.*.properties\" --no-fallback --enable-http --enable-https --enable-url-protocols=http,https --enable-sbom --install-exit-handlers --diagnostics-mode --initialize-at-run-time=io.netty.handler.ssl.BouncyCastleAlpnSslUtils,io.netty.channel.epoll.Epoll,io.netty.channel.epoll.Native,io.netty.channel.epoll.EpollEventLoop,io.netty.channel.epoll.EpollEventArray,io.netty.channel.DefaultFileRegion,io.netty.channel.kqueue.KQueueEventArray,io.netty.channel.kqueue.KQueueEventLoop,io.netty.channel.kqueue.KQueue,io.netty.channel.kqueue.Native,io.netty.channel.unix.Errors,io.netty.channel.unix.IovArray,io.netty.channel.unix.Limits,io.netty.util.internal.logging.Log4JLogger,io.netty.incubator.channel.uring -H:+BuildReport -H:ConfigurationFileDirectories=/Users/jules/workspace/myproject/modules/cli/src/main/resources/META-INF/native-image io.myproject.cli.CLI /Users/jules/workspace/myproject/cli"
# Diagnostics mode enabled: image-build reports are saved to reports/diagnostics_20230718_162020
Error: Unrecognized option: --exclude-config=.*.jar,.*.properties
```

**Comment by carlosedp** [comment]
Also the `--exclude-config` is available since Graal 23.0.0 (https://github.com/oracle/graal/pull/3179) and looks like you are using 20.0.1.

**Comment by fniephaus** [comment]
> :thinking:

Evaluating command strings in bash is always prone to escaping issues. Have you tried `--exclude-config ".*\.jar,.*\.properties"`?

```bash
$ native-image --exclude-config ".*\.jar" -m jdk.httpserver
========================================================================================================================
GraalVM Native Image: Generating 'jdk.httpserver' (executable)...
```

> Also the --exclude-config is available since Graal 23.0.0 (https://github.com/oracle/graal/pull/3179) and looks like you are using 20.0.1.

@carlosedp it seems you are confusing GraalVM version with Java version. "GraalVM for JDK 20.0.1" is internally built from the 23.0.0 sources, the GraalVM version which we are phasing out as part of the alignment with OpenJDK. So 17.0.7 and 20.0.1 should both support `--exclude-config`.

Hope this helps!

**Comment by guizmaii** [comment]
> Also the --exclude-config is available since Graal 23.0.0 (https://github.com/oracle/graal/pull/3179) and looks like you are using 20.0.1.

I'm using Oracle GraalVM 20.0.1, which is the latest version

**Comment by guizmaii** [comment]
> Evaluating command strings in bash is always prone to escaping issues. Have you tried --exclude-config ".*\.jar,.*\.properties"?

Escaping this way seems to solve the issue:
```
bash -c "native-image -cp /Users/jules/workspace/myproject/modules/cli/target/native-image-internal/manifest.jar --exclude-config \".*.jar,.*.properties\" --no-fallback --enable-http --enable-https --enable-url-protocols=http,https --enable-sbom --install-exit-handlers --diagnostics-mode --initialize-at-run-time=io.netty.handler.ssl.BouncyCastleAlpnSslUtils,io.netty.channel.epoll.Epoll,io.netty.channel.epoll.Native,io.netty.channel.epoll.EpollEventLoop,io.netty.channel.epoll.EpollEventArray,io.netty.channel.DefaultFileRegion,io.netty.channel.kqueue.KQueueEventArray,io.netty.channel.kqueue.KQueueEventLoop,io.netty.channel.kqueue.KQueue,io.netty.channel.kqueue.Native,io.netty.channel.unix.Errors,io.netty.channel.unix.IovArray,io.netty.channel.unix.Limits,io.netty.util.internal.logging.Log4JLogger,io.netty.incubator.channel.uring -H:+BuildReport -H:ConfigurationFileDirectories=/Users/jules/workspace/myproject/modules/cli/src/main/resources/META-INF/native-image io.myproject.cli.CLI /Users/jules/workspace/myproject/cli"
```

Thanks 🙂

**Comment by carlosedp** [comment]
> > Also the --exclude-config is available since Graal 23.0.0 (#3179) and looks like you are using 20.0.1.
> 
> I'm using Oracle GraalVM 20.0.1, which is the latest version

Got it.. it's the new Graal version scheme. Sorry.
